src/Vehicle/Vehicle.cc

```diff
@@ -1728,8 +1728,38 @@ void Vehicle::_handlePing(LinkInterface* link, mavlink_message_t& message)
     sendMessageOnLink(link, msg);
 }
 
+void Vehicle::_handleACEHeartbeat(mavlink_message_t& message)
+{
+   int newACECustomModeIdx = (int)mavlink_msg_heartbeat_get_custom_mode(&message);
+   QString newACEMode;
+   switch(newACECustomModeIdx) {
+       case 0: newACEMode = "ACE:Idle";    break;
+       case 1: newACEMode = "ACE:Takeoff"; break;
+       case 2: newACEMode = "ACE:Track";   break;
+       case 3: newACEMode = "ACE:RTB";     break;
+       case 4: newACEMode = "ACE:Land";    break;
+       case 5: newACEMode = "ACE:Wingman"; break;
+       default: newACEMode = QString("ACE: ") + QString::number(newACECustomModeIdx); break;
+   }
+
+   if(newACEMode != _aceMode) { //New ACE mode
+       _aceMode = newACEMode;
+       //If currently in OFFBOARD, emit a new flight mode
+       QString flight_mode = _firmwarePlugin->flightMode(_base_mode, _custom_mode);
+       if(flight_mode.compare("offboard", Qt::CaseInsensitive) == 0) {
+           emit flightModeChanged(flightMode());
+       }
+   }
+}
+
 void Vehicle::_handleHeartbeat(mavlink_message_t& message)
 {
+    //Check for an ACE message
+    if(message.compid == 41) { //Anafi ACE supervisor
```

Comment:
I use this component ID for inspecting heartbeats as well in the camera stream blocking case. I wonder if it should be captured in a static variable or something.

Reply:
Yeah I thought this would be included since its part of the mavlink messages, but it didn't work. I didn't dig.

Reply:
weird, on `planck_ctrl` side we use it from `planckAero.h` mavlink header 
```
/** @brief Planck system and component IDs */
#ifndef HAVE_ENUM_PLANCK_COMP_IDS
#define HAVE_ENUM_PLANCK_COMP_IDS
typedef enum PLANCK_COMP_IDS
{
   PLANCK_BLACKBOX_COMP_ID=32, /*  | */
   PLANCK_GPS_COMP_ID=34, /*  | */
   PLANCK_ANAFI_SUP_COMP_ID=41, /*  | */
   PLANCK_COT_COMP_ID=69, /*  | */
   PLANCK_CTRL_COMP_ID=90, /*  | */
   PLANCK_COMP_IDS_ENUM_END=91, /*  | */
} PLANCK_COMP_IDS;
#endif
```